### PR TITLE
Create HighlightTrackerServices.bambda

### PR DESCRIPTION
**Pull request by BugBountyzip**
Ever feel like you're being followed online? WhoSpyOnMe is here to confirm your suspicions!

### Bambda Contributions

* [x] Bambda has a valid [header](https://github.com/PortSwigger/bambdas/blob/73077e7ff3f6fac9db7dc95c0a00bd842b6bb64c/Proxy/HTTP/FilterOnCookieValue.bambda#L1-L5), featuring an `@author` annotation and suitable description
* [x] Bambda compiles and executes as expected
* [x] Only .bambda files have been added or modified (README.md files are automatically updated / generated after PR merge)

### CODE REVIEW

File: Proxy/HTTP/WhoSpyOnMe?.bambda [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/**
+ * WhoSpyOnMe: Burp Suite Bambda for Identifying Tracking Services
+ * FilterOut Burp Suite history to detect and analyze tracking services from web requests
+ * Capable of filtering based on Hosts, endpoints and parameters for detailed insights
+ * Useful for privacy audits and identifying potential data privacy concerns
+ * Future extension plan: 'DontSpyOnMe' to automatically block identified tracking hosts
+ * Contributions welcome: Feel free to contribute and add more hosts for detection
+ * Author: Tur24Tur / BugBountyzip (https://github.com/BugBountyzip)
```
Comment (PortSwiggerWiener) [inline Proxy/HTTP/WhoSpyOnMe?.bambda:8]: Please can you change `Author` to `@author`

File: Proxy/HTTP/WhoSpyOnMe?.bambda [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ **/
+
+// Define hash sets for hosts, paths, and parameters
+Set<String> trackedHosts = new HashSet<>(Arrays.asList("www.gstatic.com", "events.statsigapi.net", "ingesteer.services-prod.nsvcs.net", "js-eu1.hs-analytics.net", "static.hotjar.com", "forms-eu1.hscollectedforms.net", "www.google-analytics.com", "www.googletagmanager.com", "static.xx.fbcdn.net", "stats.g.doubleclick.net", "collector.github.com"));
+Set<String> trackedPaths = new HashSet<>(Arrays.asList("/logging/v1", "/track"));
+Set<String> trackedParameters = new HashSet<>(Arrays.asList("logs", "log"));
+
+// Main logic of the Bambda
+if (requestResponse.request().url() != null && requestResponse.hasResponse()) {
```
Comment (PortSwiggerWiener) [inline Proxy/HTTP/WhoSpyOnMe?.bambda:17]: Do you need this if statement? You're not using the response anywhere and can a request not have an associated URL?

File: Proxy/HTTP/WhoSpyOnMe?.bambda [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    var request = requestResponse.request();
+    var response = requestResponse.response();
```
Comment (PortSwiggerWiener) [inline Proxy/HTTP/WhoSpyOnMe?.bambda:19]: You don't need the response variable

**Review by PortSwiggerWiener — CHANGES_REQUESTED** [review]
Thanks for the submission. I've added some comments.

Please can you also rename this to be more descriptive. Ideally it would explain the function of the bambda without requiring the code to be read.

File: Proxy/HTTP/WhoSpyOnMe?.bambda (hunk 1 above)
Comment (BugBountyzip) [reply Proxy/HTTP/WhoSpyOnMe?.bambda:8]: Thank you , you are right

File: Proxy/HTTP/WhoSpyOnMe?.bambda (hunk 2 above)
Comment (BugBountyzip) [reply Proxy/HTTP/WhoSpyOnMe?.bambda:17]: Sure I agree with you Thank you

File: Proxy/HTTP/WhoSpyOnMe?.bambda (hunk 3 above)
Comment (BugBountyzip) [reply Proxy/HTTP/WhoSpyOnMe?.bambda:19]: Thank you for your support

**Comment by BugBountyzip** [comment]
i changed the name to HighlightTrackerServices but feel free to select any name you love

**Review by PortSwiggerWiener — APPROVED** [review]
Thanks for making the suggested changes.

Looks good 👍

**Review by Michelle-PortSwigger — APPROVED** [review]
👍

**Pull request merged**